backend/src/ledgers/data.ts

```diff
@@ -83,22 +98,29 @@ export async function updateLedgers() {
         console.error("BigQuery error", err);
         pagingToken = CURSOR_NOW;
       }
-      await redisClient.del(REDIS_LEDGER_KEYS[interval]);
+      await redisClient.del(REDIS_LEDGER_KEY);
     } else {
       pagingToken =
-        (await redisClient.get(REDIS_PAGING_TOKEN_KEY[interval])) || CURSOR_NOW;
+        (await redisClient.get(REDIS_PAGING_TOKEN_KEY_VALUE)) || CURSOR_NOW;
     }
 
+    console.log(
+      `${
+        isTestnet ? "[TESTNET]" : "[MAINNET]"
+      }: Starting ledgers catchup. This might take several minutes.`,
+    );
+
     await catchup(
-      REDIS_LEDGER_KEYS[interval],
+      REDIS_LEDGER_KEY,
       pagingToken,
-      REDIS_PAGING_TOKEN_KEY[interval],
+      REDIS_PAGING_TOKEN_KEY_VALUE,
       0,
       interval,
+      isTestnet,
     );
   }
 
-  const horizon = new stellarSdk.Server("https://horizon.stellar.org");
+  const horizon = new stellarSdk.Server(horizonServer);
   horizon
     .ledgers()
     .cursor(CURSOR_NOW)
```

Comment:
we need to create the right paging token key and ledger key for redis in this onmessage block, otherwise the testnet data will update the mainnet data. This should work:
```
  const horizon = new stellarSdk.Server(horizonServer);
  horizon
    .ledgers()
    .cursor(CURSOR_NOW)
    .limit(200)
    .stream({
      onmessage: async (ledger: LedgerRecord) => {
        for (const interval of intervalTypes) {
          const REDIS_LEDGER_KEY = getServerNamespace(
            REDIS_LEDGER_KEYS[interval],
            isTestnet,
          );

          const REDIS_PAGING_TOKEN_KEY_VALUE = getServerNamespace(
            REDIS_PAGING_TOKEN_KEY[interval],
            isTestnet,
          );
          await updateCache(
            [ledger],
            REDIS_LEDGER_KEY,
            REDIS_PAGING_TOKEN_KEY_VALUE,
            interval,
          );
        }
        console.log(
          `${isTestnet ? "[TESTNET]" : "[MAINNET]"}: updated to ledger ${
            ledger.closed_at
          }`,
        );
      },
    });
```

also lets add this log so we know what the latest ledger is in the logs